src/main/java/com/springboot/star_wars_character_popularity/app/controller/VoteController.java

```diff
@@ -4,9 +4,17 @@
 import com.springboot.star_wars_character_popularity.app.service.VoteService;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
 import org.springframework.http.ResponseEntity;
+import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.servlet.ModelAndView;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.List;
 
 @RestController
```

Comment:
saveVote should accept multipart form request

	@PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
	@Operation(summary = "Creates new Document")
	@ApiResponses({
			@ApiResponse(responseCode = "200", description = "Document DTO"),
			@ApiResponse(responseCode = "400", description = "Validation error : invalid argument"),
			@ApiResponse(responseCode = "500", description = "Internal Server Error"),
	})
	public DocumentDto createDocument(@Valid @Parameter(description = "Project id") @PathVariable("projectId") Long projectId,
									  @Parameter(description = "file") @RequestPart(value = "file", required = false) MultipartFile file,
									  @Parameter(description = "document meta data") @RequestPart("document") String documentRequestJson) {
		final User user = ContextService.getLoggedUser().orElseThrow();
		final DocumentCreateRequest createRequest = Util.readValue(documentRequestJson, DocumentCreateRequest.class);
		createRequest.setProjectId(projectId);
		return DocumentMapper.INSTANCE.toDto(documentService.createDocument(createRequest, user, file));
	}

Reply:
### upload document
POST http://localhost:6868/api/project/1/document
Authorization: Bearer {{accessToken}}
Content-Type: multipart/form-data; boundary=WebAppBoundary

--WebAppBoundary
Content-Disposition: form-data; name="document"
Content-Type: application/json

{
  "name": "test document with šđčćž",
  "type": "FILE",
  "source": "COMPUTER"
}

--WebAppBoundary
Content-Disposition: form-data; name="documentForm"
Content-Type: application/x-www-form-urlencoded

name=test+document&type=FILE&source=COMPUTER

--WebAppBoundary
Content-Disposition: form-data; name="file"; filename="download.jpg"
Content-Type: image/jpeg

< /home/bane/Pictures/download.jpg
--WebAppBoundary--

Reply:
Done.